ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx

```diff
@@ -0,0 +1,220 @@
+    const mappingsDict: Vector<Dictionary> | null = table.getChild(FIELD_MAPPING_FILE);
+    const mappings =
+      mappingsDict?.data
+        .map(mapping => {
+          if (mapping.dictionary == null) {
+            return [];
+          }
+          const len = mapping.dictionary.length;
+          const entries: string[] = [];
+          for (let i = 0; i < len; i++) {
+            entries.push(getLastItem(mapping.dictionary.get(i)) ?? '');
+          }
+          return entries;
+        })
+        .flat() ?? [];
+
+    // We add a EVERYTHING ELSE mapping to the list.
+    mappings.push('');
+
+    // We look through the function names to find out if there's a runtime function.
+    // Again, we only read through the dictionary, which is much faster than reading through all the rows.
+    // We stop as soon as we find a runtime function.
+    const functionNamesDict: Vector<Dictionary> | null = table.getChild(FIELD_FUNCTION_NAME);
+    functionNamesDict?.data.forEach(fn => {
+      if (fn.dictionary == null) {
+        return;
+      }
+      const len = fn.dictionary.length;
+      for (let i = 0; i < len; i++) {
+        const fn: string | null = functionNamesDict?.get(i);
+        if (fn?.startsWith('runtime') === true) {
+          mappings.push('runtime');
+          break;
+        }
+      }
+    });
+
+    // We sort the mappings alphabetically to make sure that the order is always the same.
+    mappings.sort((a, b) => a.localeCompare(b));
+    return mappings;
+  }, [table]);
+
+  // TODO: Somehow figure out how to add runtime to this, if stacks are present.
+  // Potentially read the function name dictionary and check if it contains strings starting with runtime.
+  const mappingFeatures = useMemo(() => {
+    return mappings.map(mapping => extractFeature(mapping));
+  }, [mappings]);
+
+  // TODO: Unify with mappingFeatures
+  const mappingColors = useMemo(() => {
+    const colors = {};
+    Object.entries(mappingFeatures).forEach(([_, feature]) => {
+      colors[feature.name] = getColorForFeature(feature.name, isDarkMode, colorProfile);
+    });
+    return colors;
+  }, [colorProfile, isDarkMode, mappingFeatures]);
+
+  useEffect(() => {
+    if (ref.current != null) {
+      setHeight(ref?.current.getBoundingClientRect().height);
+    }
+  }, [width]);
+
+  const xScale = useMemo(() => {
+    if (width === undefined) {
+      return () => 0;
+    }
+    return scaleLinear([0n, total], [0, width]);
+  }, [total, width]);
+
+  if (table.numRows === 0 || width === undefined) {
+    return <></>;
+  }
+
+  return (
+    <div onMouseLeave={() => dispatch(setHoveringNode(undefined))}>
+      <select
+        className="rounded-md border bg-gray-50 px-3 py-2 text-sm shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-1 focus:ring-indigo-500 dark:bg-gray-900"
+        onChange={e => setSortBy(e.target.value)}
+      >
+        <option value="function_name">Function Name</option>
```

Comment:
Shall we place the select dropdown next to the `Reset View` button and add a text in front. Like this:

"Reset View"  "Sort by: `<Select />`"